src/policy/mod.rs

```diff
@@ -124,8 +124,8 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Liftable<Pk> for Miniscript<Pk, Ctx>
 impl<Pk: MiniscriptKey, Ctx: ScriptContext> Liftable<Pk> for Terminal<Pk, Ctx> {
     fn lift(&self) -> Result<Semantic<Pk>, Error> {
         let ret = match *self {
-            Terminal::PkK(ref pk) => Semantic::KeyHash(pk.to_pubkeyhash()),
-            Terminal::PkH(ref pkh) => Semantic::KeyHash(pkh.clone()),
+            Terminal::PkK(ref pk) | Terminal::PkH(ref pk) => Semantic::KeyHash(pk.to_pubkeyhash()),
+            Terminal::RawPkH(ref pkh) => Semantic::KeyHash(pkh.clone()),
```

Comment:
In 36ff0d2bf8b9df789a5581f6a10f9f4457803331:

I guess this is for a future PR, but I'd kinda like to make `semantic::Policy` have only `Pk`s, not `Pk::Hash`es, and then refuse to lift any scripts that had `RawPkH` in it. We could provide a transformer that took a `Miniscript<Pk>` to a `Miniscript<Pk::Hash>` maybe, converting `RawPkH`s to `PkH`s along the way, to get the old behavior.

Reply:
Yup, the future TODOs are listed https://github.com/rust-bitcoin/rust-miniscript/pull/431#issuecomment-1157153192